index.js

```diff
@@ -24,6 +28,62 @@ const pkg = require('./package');
 const getMethodRedirectCodes = new Set([300, 301, 302, 303, 304, 305, 307, 308]);
 const allMethodRedirectCodes = new Set([300, 303, 307, 308]);
 
+const isFormData = body => isStream(body) && typeof body.getBoundary === 'function';
+
+const getBodySize = opts => {
+	const body = opts.body;
+
+	return new Promise((resolve, reject) => {
+		if (opts.headers['content-length']) {
+			resolve(Number(opts.headers['content-length']));
+			return;
+		}
+
+		if (!body && !opts.stream) {
+			resolve(0);
+			return;
+		}
+
+		if (typeof body === 'string') {
+			resolve(Buffer.byteLength(body));
+			return;
+		}
+
+		if (isFormData(body)) {
+			body.getLength((err, size) => {
```

Comment:
Would be nicer to use `pify` here and on line 66, so we don't have to wrap the whole thing in the Promise constructor.